### PR TITLE
[14.0][REF] hr_employee_firstname,hr_employee_lastnames: Removed partner-contact dependencies

### PR DESCRIPTION
**Pull request by luiseevaquer**
This change makes the module work without the need to install the functionalities in partner, making the process independent of the [partner-contact](https://github.com/OCA/partner-contact/tree/14.0) modules ([partner_fistname](https://github.com/OCA/partner-contact/tree/14.0/partner_firstname) and [partner_second_lastname](https://github.com/OCA/partner-contact/tree/14.0/partner_second_lastname) respectively)

Demostrative Video:  https://drive.google.com/file/d/1Rllj1pKQwxC6J4hJdHTXgB4pzw3AA2fM/view

### CODE REVIEW

**Comment by luiseevaquer** [comment]
Hi @luistorresm 

What do you think?

**Comment by luistorresm** [comment]
LGTM @lescobarvx 

I will to tests,

Could you fix the CI and complete the PR description, please?

Regards

**Review by luistorresm — APPROVED** [review]
Thanks @lescobarvx

File: hr_employee_firstname/models/hr_employee.py [hunk 1]
```diff
@@ -9,15 +9,34 @@
...
+
+    @api.model
+    def _get_names_order(self):
+        """Get names order configuration from system parameters.
+        You can override this method to read configuration from language,
+        country, company or other"""
+        return (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("employee_names_order", self._names_order_default())
```
Comment (hugho-ad) [inline hr_employee_firstname/models/hr_employee.py:30]: Could you improve indentation please, remember min length now it's 119

File: hr_employee_firstname/models/hr_employee.py [hunk 2]
```diff
@@ -65,15 +81,58 @@ def _prepare_vals_on_write_firstname_lastname(self, vals):
...
+        Removes leading, trailing and duplicated whitespace.
+        """
+        try:
+            name = " ".join(name.split()) if name else name
+        except UnicodeDecodeError:
+            name = " ".join(name.decode("utf-8").split()) if name else name
+
+        if comma:
+            name = name.replace(" ,", ",")
+            name = name.replace(", ", ",")
```
Comment (hugho-ad) [inline hr_employee_firstname/models/hr_employee.py]:
```suggestion
            name = name.replace(", ", ",").replace(" ,", ",")
```

Comment (luiseevaquer) [reply hr_employee_firstname/models/hr_employee.py]: Done

File: hr_employee_firstname/models/hr_employee.py (hunk 1 above)
Comment (luiseevaquer) [reply hr_employee_firstname/models/hr_employee.py:30]: When I try to make this change, Travis CI shows the following error proposing as it was previously:

Selección_925


So I reversed the change

File: hr_employee_firstname/models/hr_employee.py [hunk 3]
```diff
@@ -92,14 +150,6 @@ def _install_employee_firstname(self):
         records._inverse_name()
         _logger.info("%d employees updated installing module.", len(records))
 
-    def _update_partner_firstname(self):
```
Comment (astirpe) [inline hr_employee_firstname/models/hr_employee.py]: Here you are removing part of the functionality of the module. You could keep this logic by checking whether the contact_firstname module is installed. (Please avoid yet another glue module...)

**Review by astirpe — CHANGES_REQUESTED** [review]

File: hr_employee_firstname/models/hr_employee.py (hunk 1 above)
Comment (luistorresm) [reply hr_employee_firstname/models/hr_employee.py:30]: @hugho-ad

The OCA guidelines need this structure and are based on the partner_contact module, I prefer this structure

Screen Shot 2020-10-29 at 9 04 27

**Comment by astirpe** [comment]
Travis is red

**Review by astirpe — APPROVED** [review]
Code LGTM. Not tested.

**Review by hugho-ad — APPROVED** [review]

**Comment by luistorresm** [comment]
Hello @moylop260 

Could you review this PR, and merge if you agree, please?

Regards

**Review by moylop260 — APPROVED** [review]

**Comment by moylop260** [comment]
Hi @sbidoul 
I think it is a error with prettier

May I ask you to check it, please?

**Comment by sbidoul** [comment]
@moylop260 this is https://github.com/OCA/oca-addons-repo-template/pull/38.

**Pull request merged**